src/Any/Signer.cpp

```diff
@@ -74,6 +77,57 @@ Any::Proto::SigningOutput Any::Signer::sign() const noexcept {
             }
             break;
         }
+        case TWCoinTypeIoTeX: {
+            IoTeX::Proto::SigningInput message;
+            parse(transaction, &message, output);
+            if (output.success()) {
+                message.set_privatekey(privateKey.bytes.data(), privateKey.bytes.size());
+                auto signerOutput = IoTeX::Signer(std::move(message)).build();
+                auto encoded = signerOutput.encoded();
+                output.set_output(hex(encoded.begin(), encoded.end()));
+            }
+            break;
+        }
+        case TWCoinTypeWanchain: {
+            Ethereum::Proto::SigningInput message;
+            parse(transaction, &message, output);
+            if (output.success()) {
+                auto ethTransaction = Ethereum::Transaction(
+                        /* nonce: */ load(message.nonce()),
+                        /* gasPrice: */ load(message.gas_price()),
+                        /* gasLimit: */ load(message.gas_limit()),
+                        /* to: */ Ethereum::Address(message.to_address()),
+                        /* amount: */ load(message.amount()),
+                        /* payload: */ Data(message.payload().begin(), message.payload().end())
+                );
+                auto signer = Wanchain::Signer(load(message.chain_id()));
+                signer.sign(privateKey, ethTransaction);
+                auto encoded = signer.encode(ethTransaction);
+                output.set_output(hex(encoded.begin(), encoded.end()));
+            }
+            break;
+        }
+        case TWCoinTypeWaves: {
+            Waves::Proto::SigningInput message;
+            parse(transaction, &message, output);
+            if (output.success()) {
+                auto publicKey = privateKey.getPublicKey(TWPublicKeyTypeCURVE25519);
+                auto wavesTransaction = Waves::Transaction(
+                        /* amount */ message.amount(),
+                        /* asset */ message.asset(),
+                        /* fee */ message.fee(),
+                        /* fee_asset */ message.fee_asset(),
+                        /* to */ Waves::Address(message.to()),
+                        /* attachment */ Data(message.attachment().begin(), message.attachment().end()),
+                        /* timestamp */ message.timestamp(),
+                        /* pub_key */ publicKey.bytes
```

Comment:
not related to AnySigner, would it be possible to remove publicKey from the transaction? and inject it inside Waves Signer? it would be much easier to operate, as we do for most proto bufs where pub key is encapsulated and you only pass private key and pub key would be derived inside the signer

Reply:
@hewigovens what's the best practice should be here for public keys? it's a separate issue tho, but would be great to address

Reply:
we can remove it if we can get public key from private key, but some cases it might be other public keys